pkg/controller/servicebindingrequest/binder_test.go

```diff
@@ -36,24 +39,92 @@ func getEnvVar(envVars []corev1.EnvVar, name string) *corev1.EnvVar {
 	return nil
 }
 
+func TestBindingCustomSecretPath(t *testing.T) {
+	ns := "custombinder"
+	name := "service-binding-request-custom"
+	matchLabels := map[string]string{
+		"appx": "x",
+	}
+
+	f := mocks.NewFake(t, ns)
+	sbrSecretPath := f.AddMockedServiceBindingRequest(name, &ns, "ref-custom-podspec", "deployment", deploymentsGVR, matchLabels)
+	f.AddMockedUnstructuredDeployment("deployment", matchLabels)
+
+	customSecretPath := "metadata.clusterName"
+	sbrSecretPath.Spec.ApplicationSelector.BindingPath = &v1alpha1.BindingPath{
+		PodSpecPath: &v1alpha1.PodSpecPath{
+			Containers: v1alpha1.DefaultPathToContainers,
+			Volumes:    v1alpha1.DefaultPathToVolumes,
+		},
+		CustomSecretPath: &customSecretPath,
+	}
+	binderForsbrSecretPath := newBinder(
+		context.TODO(),
+		f.FakeDynClient(),
+		sbrSecretPath,
+		[]string{},
+		testutils.BuildTestRESTMapper(),
+	)
+	require.NotNil(t, binderForsbrSecretPath)
+
+	t.Run("custom secret field path", func(t *testing.T) {
+		secretPath := binderForsbrSecretPath.getSecretFieldPath()
+		expectedSecretPath := []string{"metadata", "clusterName"}
+		require.True(t, reflect.DeepEqual(secretPath, expectedSecretPath))
+	})
+
+	t.Run("update custom secret field path ", func(t *testing.T) {
+		list, err := binderForsbrSecretPath.search()
+		require.NoError(t, err)
+		require.Equal(t, 1, len(list.Items))
+
+		updatedDeployment, err := binderForsbrSecretPath.updateSecretField(&list.Items[0])
+		require.NoError(t, err)
+		require.NotNil(t, updatedDeployment)
+
+		customSecretPathSlice := strings.Split(customSecretPath, ".")
+
+		customSecretInMeta, found, err := unstructured.NestedFieldCopy(list.Items[0].Object, customSecretPathSlice...)
+		require.NoError(t, err)
+		require.True(t, found)
+		require.Equal(t, name, customSecretInMeta)
+	})
+}
+
 func TestBinderNew(t *testing.T) {
 	ns := "binder"
 	name := "service-binding-request"
 	matchLabels := map[string]string{
 		"connects-to": "database",
 		"environment": "binder",
 	}
+
 	f := mocks.NewFake(t, ns)
 	sbr := f.AddMockedServiceBindingRequest(name, nil, "ref", "", deploymentsGVR, matchLabels)
+	sbr.Spec.ApplicationSelector.BindingPath = &v1alpha1.BindingPath{
```

Comment:
Defaults are set in `SetDefaults`, so I believe this assignment might not be necessary anymore.

Reply:
I tried to remove it. But that seems to be not working.

Reply:
It means that `SetDefaults` might not be doing what it is expected from it, as [commented here](https://github.com/redhat-developer/service-binding-operator/pull/513/files#r459312216).

Reply:
The current code is doing what you explained. The default values are set in `SetDefaults`. There is still a possibility that the application selector could be `nil`. So, when accessing it should be checked for `nil`. Similarly `BindingPath` and `PodSpecPath` also could be `nil`.

Reply:
Done. 25214178b6bbf0677d498ce522a7d1c595f63c9f